esphome/components/output/switch/output_switch.h

```diff
@@ -19,6 +30,7 @@ class OutputSwitch : public switch_::Switch, public Component {
   void write_state(bool state) override;
 
   output::BinaryOutput *output_;
+  OutputSwitchRestoreMode restore_mode_{OUTPUT_SWITCH_RESTORE_DEFAULT_OFF};
```

Comment:
No need for the default here as codegen will always call `set_restore_mode`
```suggestion
  OutputSwitchRestoreMode restore_mode_;
```